datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/prim_op.rs

```diff
@@ -43,8 +46,8 @@ where
     T: ArrowPrimitiveType + Send,
     F: Fn(&mut T::Native, T::Native) + Send + Sync,
 {
-    /// values per group, stored as the native type
-    values: Vec<T::Native>,
+    /// Values per group, stored as the native type
```

Comment:
Something I noticed several times was `VecDeqeue<B:BlocK>` being used -- did you consider making a special struct for this? Something like

```rust
struct Blocks<B: Block> { 
 inner: VecDeque<Block>
}

impl Blocks<B: Block> {
  // resize filling all to default value
  fn resize(&mut self, num_values: usize, default_value: Block::T) { .. }
  
  fn ensure_room_enough_for_blocks(...) 
}
```

That would make it easier to understand and document how lists of blocks are managed and would make it easier to read code that used it. It would also make `ensure_room_enough_for_blocks` more discoverable

You might also be able to encapsulate the `block_size` parameter in `Blocks` which would make migrating other GroupsAccumulators easier

Reply:
Addressed.
I have defined `Blocks` in a dedicated moduel `blocks.rs`, and simplify all related codes using it.